[WIP] Revisit sypgp to enable better testing coverage

gvallee opened the pull request:
**Description of the Pull Request (PR):**

Improve testing coverage for sypgp package. Addresses issue #2559

**This fixes or addresses the following GitHub issues:**

- Addresses #2559 


**Before submitting a PR, make sure you have done the following:**

- Read the [Guidelines for Contributing](https://github.com/sylabs/singularity/blob/master/CONTRIBUTING.md), and this PR conforms to the stated requirements.
- Added changes to the [CHANGELOG](https://github.com/sylabs/singularity/blob/master/CHANGELOG.md) if necessary according to the [Contribution Guidelines](https://github.com/sylabs/singularity/blob/master/CONTRIBUTING.md)
- Added tests to validate this PR and tested this PR locally with a `make testall`
- Based this PR against the appropriate branch according to the [Contribution Guidelines](https://github.com/sylabs/singularity/blob/master/CONTRIBUTING.md)
- Added myself as a contributor to the [Contributors File](https://github.com/sylabs/singularity/blob/master/CONTRIBUTORS.md)


Attn: @singularity-maintainers

bauerm97 committed:
Switch to using go modules

- Add go.mod (and the corresponding go.sum)
- Fix makeit/gengodep so that it lists all the go files that are
  needed to build a particular import path
- Add internal/pkg/runtime/engines/singularity/plugins.go to have
  something that pulls the relevant plugin code.
- Remove Gopkg.toml and Gopkg.lock
- Updated vendor/ with go mod vendor
- Make sure we are calling the same go compiler in all cases by
  resolving the absolute path only once.
- Explicitly export GO111MODULE=on to make sure that everything is
  running with module support activated. In particular, this seems
  necessary to build the CNI plugins, as having a go.mod file and
  passing -mod=vendor in that case doesn't seem to be sufficient (go
  still fails to locate the relevant modules).
- Replace dep with go mod

Signed-off-by: Michael Bauer <bauerm@umich.edu>
Signed-off-by: Marcelo E. Magallon <marcelo@sylabs.io>

cclerget committed:
First enhancement for image package

cclerget committed:
Add image section/partition reader helper

cclerget committed:
Add static error for comparaison

Commit:
Print usage information if an invalid flag is passed to singularity

In cobra, rootCmd.Execute() processes the command line and executes the
appropriate command or subcommand. If an invalid flag is passed to the
subcommand, the usage information for the subcommand is printed. If an
invalid subcommand or flag is passed to the root command, the usage
information for the root command is printed. If in invalid subcommand is
passed along with an invalid flag (`singularity -f xxx shell`), cobra
prints an error. Since we are suppressing cobra errors (SilenceErrors),
nothing is printed and singularity returns silently.

Fix this by passing SilenceUsage as well, and take care of printing the
usage string ourselves for all errors.

- Fixes: #2534

Signed-off-by: Marcelo E. Magallon <marcelo@sylabs.io>

cclerget committed:
Add squashfs unpacker

cclerget committed:
Add section/partition reader tests

cclerget committed:
Remove code redundancy in section reader. Remove useless sections from SIF image. Fix comment oversight and cleanup code a bit.

cclerget committed:
More descriptive error message

cclerget committed:
Fix Image structure comment

Commit:
Prealloc slices where possible

Code like this:

    var names []string
    for n := range c.Remotes {
        names = append(names, n)
    }

knows in advance the required capacity for the slice, so it's possible
to avoid trashing in the allocator by simply preallocating the slice,
like this:

    names := make([]string, 0, len(c.Remotes))
    for n := range c.Remotes {
        names = append(names, n)
    }

Signed-off-by: Marcelo E. Magallon <marcelo@sylabs.io>

jmstover committed:
Remove plugin import

jmstover committed:
Fix imports

jmstover committed:
Remove old internal/pkg/image/* files that didn't get removed

GodloveD committed:
Merge pull request #2917 from jmstover/wip_release_merge

3.1.1 release merge

Commit:
Updated helps + typos

Commit:
Saving point; WIP

jmstover committed:
Fix RPM spec building on SLES 11 (#2942)

* Fix RPM spec building on SLES 11

Go 1.8.3 is the newest Go version availible in the repositories so
we remove Go as a BuildRequire and expect it to be in the PATH
when rpmbuild is ran

libseccomp1 is to old to be used. It actually cannot be installed
as currently makeit will pick it up as existing and try using it.
Remove is as a BuildRequire as well.

* Try to catch undefine value

* Another attempt at making macros happy

* Okay... maybe my quoting was incorrect somehow when I first tried this way.

Commit:
fixed typo on cache help command

gvallee committed:
fixes #2927; refactor singularity commands test

Commit:
Updated --sumary flag to be compatible with --type=

Commit:
Merge remote-tracking branch 'upstream/master' into cache-cmd

Commit:
%v => %d

Commit:
Updated listOciCache comment

GodloveD committed:
Merge pull request #2961 from devprincess/typocachehelp_3.1

Fixed typo on cache help command -> release 3.1

Commit:
Updated funcs comments

Commit:
Use PACKAGE_NAME in --version output

Commit:
Added 'key remove' command

Commit:
Updated docs

Commit:
Implement plugin inspect command

- Fixes #2323

Signed-off-by: Marcelo E. Magallon <marcelo@sylabs.io>

Commit:
Removed extra var

cclerget committed:
Read link from main thread (#3022)

cclerget committed:
Change directory from main thread with sandbox images (#3023)

Commit:
Updated tests

Commit:
Opps! wrong branch!

This reverts commit b57cccce0710566419bb7b406e410c49f0913f06.

Commit:
Merge remote-tracking branch 'upstream/master' into key-remove

DrDaveD committed:
fix error message typo (#2920)

Commit:
Updated errors, And strings, and function

cclerget committed:
Add caps alias for capability command

cclerget committed:
Fix typo while managing caps

cclerget committed:
Remove root requirement for capability list

cclerget committed:
Return when using --desc

cclerget committed:
Remove options for capability list

cclerget committed:
Fix grammar

cclerget committed:
Help refactor. Add capability avail command and move description to this one

cclerget committed:
Fix error message

cclerget committed:
capability help refactor

cclerget committed:
Fix instance list help typo

Commit:
Change message when no help is present

Commit:
Refactor singularity help run-help text

Commit:
Remove duplicated singularity

Commit:
Merge remote-tracking branch 'upstream/master' into key-remove

Commit:
Document blobSum cache option

Commit:
Fix typo

Commit:
Merge remote-tracking branch 'upstream/master' into cache-cmd

Commit:
Combined listLibraryCache(), & listOciCache() to listTypeCache()

Commit:
Removed unsued code

Commit:
Merge remote-tracking branch 'upstream/master' into key-remove

mem committed:
Use a better wording to describe cleaned cache

Co-Authored-By: sashayakovtseva <sashayakovtseva@gmail.com>

GodloveD committed:
Add user-defined cflags to library checks (#3074)

Commit:
Mention SINGULARITY_CACHEDIR in list cache command

Co-Authored-By: sashayakovtseva <sashayakovtseva@gmail.com>

Commit:
Removed "extra" cobra Args

Commit:
Merge remote-tracking branch 'upstream/master' into cache-cmd

jmstover committed:
Merge branch 'release-3.1' into 2942-cherry-pick

jmstover committed:
Merge branch 'release-3.1' into 2990-cherry-pick

jmstover committed:
Merge branch 'release-3.1' into 3022-cherry-pick

jmstover committed:
Merge branch 'release-3.1' into 3023-cherry-pick

jmstover committed:
Merge branch 'release-3.1' into 2920-cherry-pick

jmstover committed:
Merge branch 'release-3.1' into 3068-cherry-pick

jmstover committed:
Merge branch 'release-3.1' into 3069-cherry-pick

jmstover committed:
Merge branch 'release-3.1' into 3070-cherry-pick

cclerget committed:
Fix strings.Split side effect and return explicit error message while listing avail capabilities

ikaneshiro committed:
Changes to Definition struct to allow section args

	This PR changes many of the fields of the Definition struct in
	order to encapsulate the idea of a section having additional
	args. This requires some changes throughout the build package in
	order to accomodate the change. This also changes the parser to
	parse those arguements properly into the Definition struct. Also
	fixes some comments

cclerget committed:
Fix overlay directory not deleted caused by defer evaluation

cclerget committed:
Check container state before attach

Commit:
Add missing quote

cclerget committed:
Fix pause/resume with exec command and move pause/resume control in master process

jscook2345 committed:
Update the message shown when oci kill is run on a container without a state of created or running

jscook2345 committed:
Update the message shown when oci delete is run on a container without a state of created or stopped

jscook2345 committed:
Update the message shown when oci start is run on a container without a state of created

GodloveD committed:
Do remote build when builder flag is set (#3028) (#3081)

* perform remote build if --builder flag is specified

* Set remote to true in preRun() if builder is changed

	This removes preRun() from build_linux.go and build_darwin.go
	and moves it to build.go since those functions are identical.

mem committed:
Merge branch 'release-3.1' into 3022-cherry-pick

mem committed:
Merge branch 'release-3.1' into 3023-cherry-pick

GodloveD committed:
Merge branch 'release-3.1' into 2942-cherry-pick

GodloveD committed:
Merge pull request #3076 from GodloveD/2942-cherry-pick

Fix RPM spec building on SLES 11 (2942 cherry-pick)

GodloveD committed:
Gv/2938 (#2958) (#3085)

* fixes #2938; fix a typo on cache clean --help

* Add myself to the list of contributors

mem committed:
Merge branch 'release-3.1' into 3022-cherry-pick

Commit:
Gofmt file

cclerget committed:
Check container status in oci exec command

cclerget committed:
Update format string

cclerget committed:
Improve oci create error messages

cclerget committed:
Add sub directory notion for instance helpers

cclerget committed:
Fix error message for oci state

GodloveD committed:
Merge branch 'release-3.1' into 2920-cherry-pick

cclerget committed:
Add sudo in capability add/drop example

ikaneshiro committed:
Adding ability for multiple files sections

	This introduces a change to the parser that removes comments
	from lines before parsing for simpler parsing. Also expands
	definition struct to allow multiple files objects to enable
	multistage builds. Unit tests for parser updated to represent
	this struct change.

GodloveD committed:
Merge pull request #3082 from GodloveD/3022-cherry-pick

Read link from main thread (3022 cherry-pick)

DrDaveD committed:
clarify confusing rpm build instructions

ikaneshiro committed:
comment fixes

GodloveD committed:
check for existing file before building docker based continer during pull (#3075)

Commit:
Merge remote-tracking branch 'upstream/master' into cache-cmd

mem committed:
Merge branch 'master' into issue_2534

GodloveD committed:
Merge branch 'release-3.1' into 2990-cherry-pick

ikaneshiro committed:
Add All() function to parser

	All() will parse all stages in a file and return a slice
	containing definitions of those stages.

Commit:
Quick test fix (#3136)

mem committed:
Merge branch 'master' into plugin_inspect

mem committed:
Merge branch 'master' into issue_2534

GodloveD committed:
Merge pull request #3137 from GodloveD/3136-cherry-pick

Quick test fix (3136 cherry-pick)

GodloveD committed:
Merge branch 'release-3.1' into 2990-cherry-pick

GodloveD committed:
Merge pull request #3077 from GodloveD/2990-cherry-pick

Use PACKAGE_NAME in --version output (2990 cherry-pick)

GodloveD committed:
Merge branch 'release-3.1' into 3023-cherry-pick

GodloveD committed:
Merge pull request #3083 from GodloveD/3023-cherry-pick

Change directory from main thread with sandbox images (3023 cherry pick)

GodloveD committed:
Merge branch 'release-3.1' into 2920-cherry-pick

ikaneshiro committed:
Add ability to copy files between stages

GodloveD committed:
Merge pull request #3086 from GodloveD/2920-cherry-pick

Fix start.c error message typo (2920 cherry-pick)

GodloveD committed:
Merge branch 'release-3.1' into 3068-cherry-pick

GodloveD committed:
Merge pull request #3087 from GodloveD/3068-cherry-pick

Fix instance list help typo (3068 cherry pick)

GodloveD committed:
Merge branch 'release-3.1' into 3069-cherry-pick

GodloveD committed:
Merge pull request #3088 from GodloveD/3069-cherry-pick

Change message when no help is present (3069 cherry pick)

GodloveD committed:
Merge branch 'release-3.1' into 3070-cherry-pick

GodloveD committed:
Merge pull request #3090 from GodloveD/3070-cherry-pick

Refactor singularity help run-help text (3070 cherry pick)

GodloveD committed:
Merge branch 'release-3.1' into 2962-cherry-pick

GodloveD committed:
undoing changes to vendor file

GodloveD committed:
Merge branch '2962-cherry-pick' of github.com:GodloveD/singularity into 2962-cherry-pick

cclerget committed:
Remove all subcommand in short use helps

cclerget committed:
Store temporary image in /tmp instead of HOME/.singularity/cache for now. Will be implemented further with cache support

GodloveD committed:
Merge pull request #3091 from GodloveD/2962-cherry-pick

fixes #2927; refactor singularity commands test (2962 cherry pick)

ikaneshiro committed:
change new test cases to reflect changes in parser package

ikaneshiro committed:
fix gofmt error

ikaneshiro committed:
CLI integration and some edge case checking

mem committed:
Merge branch 'master' into clarify-rpm-build

ikaneshiro committed:
Merge remote-tracking branch 'upstream/master' into multistage_section_args

Commit:
Merge remote-tracking branch 'upstream/master' into key-remove

Commit:
Fixed extra long hash image pull

Commit:
Merge remote-tracking branch 'upstream/master' into cache-cmd

ikaneshiro committed:
Some e2e testing for multistage builds

	These tests are mainly to test the copying between stages.

ikaneshiro committed:
fix formatting to make linter happy

GodloveD committed:
Merge pull request #3145 from GodloveD/3110-cherry-pick

Check container state before attach (3110 cherry pick)

GodloveD committed:
Merge branch 'release-3.1' into 3109-cherry-pick

GodloveD committed:
Merge pull request #3146 from GodloveD/3109-cherry-pick

Fix overlay directory not deleted caused by defer evaluation (3109 cherry pick)

GodloveD committed:
Merge branch 'release-3.1' into 3114-cherry-pick

GodloveD committed:
Merge pull request #3147 from GodloveD/3114-cherry-pick

Fix oci pause/resume command (3114 cherry pick)

GodloveD committed:
Merge branch 'release-3.1' into 3128-cherry-pick

GodloveD committed:
Merge pull request #3148 from GodloveD/3128-cherry-pick

Improve oci create error messages (3128 cherry pick)

GodloveD committed:
Merge branch 'release-3.1' into 3121-cherry-pick

GodloveD committed:
Merge pull request #3150 from GodloveD/3121-cherry-pick

Update the message shown when oci start is run on a container without…

GodloveD committed:
Merge branch 'release-3.1' into 3120-cherry-pick

GodloveD committed:
Merge pull request #3151 from GodloveD/3120-cherry-pick

Update the message shown when oci delete is run on a container withou…

GodloveD committed:
Merge branch 'release-3.1' into 3119-cherry-pick

GodloveD committed:
Merge pull request #3152 from GodloveD/3119-cherry-pick

Update the message shown when oci kill is run on a container without …

GodloveD committed:
Merge branch 'release-3.1' into 3072-cherry-pick

GodloveD committed:
Merge pull request #3154 from GodloveD/3072-cherry-pick

Document blobSum type cache option (3072 cherry pick)

GodloveD committed:
Merge branch 'release-3.1' into 3066-cherry-pick

GodloveD committed:
Merge pull request #3155 from GodloveD/3066-cherry-pick

Capability command fixes (3066 cherry pick)

jmstover committed:
Merge branch 'release-3.1' into 3129-cherry-pick

GodloveD committed:
Merge branch 'release-3.1' into 3140-cherry-pick

Commit:
Improve sypgp tests

Commit:
Improve sypgp error handling

GodloveD committed:
Merge pull request #3149 from GodloveD/3129-cherry-pick

Add sub directory notion for instance helpers (3129 cherry pick)

GodloveD committed:
Merge branch 'release-3.1' into 3140-cherry-pick

GodloveD committed:
Merge pull request #3160 from GodloveD/3140-cherry-pick

Remove all subcommand in short use helps (3140 cherry pick)

GodloveD committed:
Merge branch 'release-3.1' into 3141-cherry-pick

GodloveD committed:
Merge pull request #3161 from GodloveD/3141-cherry-pick

Store temporary image in /tmp instead of HOME/.singularity/cache for …

cclerget committed:
Merge branch 'master' into clarify-rpm-build

gvallee committed:
Revisit the sygpg package to enable better testing coverage. Addresses issue #2559

gvallee committed:
Revisit the sygpg package to enable better testing coverage. Addresses issue #2559

ikaneshiro committed:
Merge branch 'master' into key-remove

cclerget committed:
Merge pull request #3130 from DrDaveD/clarify-rpm-build

Clarify confusing rpm build instructions

ikaneshiro committed:
Def parser fix to append multiple %files sections

	Fixes error in parser where consecutive %files sections
	were not appended. A test case has been added to catch this
	bug in the future.

ikaneshiro committed:
Merge branch 'master' into key-remove

ikaneshiro committed:
Merge pull request #3047 from WestleyK/key-remove

'key remove' Command

tri-adam committed:
Merge pull request #3162 from tri-adam/key_url_fix

Improve Key Command URL Handling

GodloveD committed:
fixed up changelog and bumped version in install

GodloveD committed:
Merge pull request #3166 from GodloveD/update-CHANGELOG.md

fixed up changelog and bumped version in install

ikaneshiro committed:
removed accidentally added file

gvallee committed:
fixes #2986. Refactor singularity help pull options text.

ikaneshiro committed:
fix comments

DrDaveD committed:
add info messages when converting from SIF and removing temporary converted image (#3172)

mem committed:
Merge branch 'master' into issue_2534

GodloveD committed:
Merge branch 'master' into plugin_inspect

GodloveD committed:
Merge pull request #3056 from mem/plugin_inspect

Implement plugin inspect command

GodloveD committed:
Merge branch 'master' into cache-cmd

GodloveD committed:
Merge pull request #2919 from WestleyK/cache-cmd

Improved Cache List Output Summery

jmstover committed:
Support Sandbox Image on macOS

Check if we have a direcotory used for the image, and if so be
sure to pass it instead of a image.

jmstover committed:
Generate 9p Mount Tag on Bind Mount

Generate a random 6 alphabetic char 9p mount tag for all mounts that
are specified with the `-B` option. This means that a user can set
both `-B /tmp -B /var/tmp` and it will work as expected.

Previously `tmp` would have been used as the mount tag for both
of the bind mounts.

ikaneshiro committed:
fixed more error messages

gvallee committed:
Start the beginning of the sentence with a lower case word for consistency

ikaneshiro committed:
return error instead of fatal call

gvallee committed:
start sentences with a low-case word for consistency

Commit:
Remove early debug output in internal/pkg/plugin

Since the init functions for internal/pkg/plugin are running before
those for sylog, the default log level is still at "debug" and these
messages show up whenever a plugin is enabled (irrespective of the
actual log level set on the command line).

Since these are not critical, remove them. If they are deemed to be
useful later, they can be restored after the underlying issue is fixed.

Fixes: #3133

Signed-off-by: Marcelo E. Magallon <marcelo@sylabs.io>

ikaneshiro committed:
Merge branch 'master' into multistage_stage_copy

ikaneshiro committed:
Merge pull request #3156 from ikaneshiro/multistage_stage_copy

Multistage Builds

jscook2345 committed:
Add an options section to the usage for the oci kill command

GodloveD committed:
Merge branch 'master' into gv/2986

GodloveD committed:
Merge pull request #3168 from gvallee/gv/2986

Refactor singularity help pull options text.

GodloveD committed:
Merge branch 'master' into remove_early_plugin_debug_messages

GodloveD committed:
Merge pull request #3179 from mem/remove_early_plugin_debug_messages

Remove early debug output in internal/pkg/plugin

GodloveD committed:
Merge branch 'master' into add-missing-usage-for-oci-kill

GodloveD committed:
Merge pull request #3183 from jscook2345/add-missing-usage-for-oci-kill

Add an options section to the usage for the oci kill command

cclerget commented on pkg/sypgp/sypgp.go:
```diff
@@ -212,6 +249,10 @@ func LoadPubKeyring() (openpgp.EntityList, error) {
 // PrintEntity pretty prints an entity entry
 func PrintEntity(index int, e *openpgp.Entity) {
+	if e == nil {
+		return
```
maybe display a warning as we don't return any error here

cclerget commented on pkg/sypgp/sypgp.go:
```diff
@@ -255,6 +296,10 @@ func PrintPrivKeyring() (err error) {
 // StorePrivKey stores a private entity list into the local key cache
 func StorePrivKey(e *openpgp.Entity) (err error) {
+	if e == nil {
+		return
```
safest to return an explicit error message

cclerget commented on pkg/sypgp/sypgp.go:
```diff
@@ -269,6 +314,10 @@ func StorePrivKey(e *openpgp.Entity) (err error) {
 // StorePubKey stores a public key entity list into the local key cache
 func StorePubKey(e *openpgp.Entity) (err error) {
+	if e == nil {
+		return
```
same here

cclerget commented on pkg/sypgp/sypgp.go:
```diff
@@ -461,6 +514,10 @@ func GenKeyPair(keyServiceURI string, authToken string) (entity *openpgp.Entity,
 // DecryptKey decrypts a private key provided a pass phrase
 func DecryptKey(k *openpgp.Entity) error {
+	if k == nil {
+		return nil
```
should return an error because this is a misuse of this function

cclerget commented on pkg/sypgp/sypgp.go:
```diff
@@ -476,11 +533,18 @@ func DecryptKey(k *openpgp.Entity) error {
 // EncryptKey encrypts a private key using a pass phrase
 func EncryptKey(k *openpgp.Entity, pass string) (err error) {
+	if k == nil {
+		return nil
```
same here

cclerget committed:
Fix warning when adding already assigned capabilities or dropping non-existent capabilities. Fix unit tests.

cclerget committed:
Check if an instance is not started from another one and display an appropriate error

ikaneshiro committed:
Change action test script to default to non-interactive behavior

	This change runs `/bin/sh -c true` instead of `/bin/sh` so that
	it is no longer interactive by default. This change is reflected
	in the container metadata actions scripts as well as the action
	scripts bound in at runtime.

Commit:
Clean up building of CNI plugins

- Remove some redundant variables and compute values from original list.
- Document the purpose and usage of
  internal/pkg/runtime/engines/singularity/plugins.go

Signed-off-by: Marcelo E. Magallon <marcelo@sylabs.io>

mem committed:
Wait for dpkg lock on CircleCI (#3192)

It seems that when the VM boots, it runs `apt-get update` (or something
similar) and this prevents `apt-get install ...` from working.

Wait until that lock is released. Since the VM is using a somewhat old
version of Ubuntu, other (less racy) methods to handle this are not
available.

Signed-off-by: Marcelo E. Magallon <marcelo@sylabs.io>

ikaneshiro committed:
Merge branch 'master' into non_interactive_test

ikaneshiro committed:
Merge pull request #3189 from ikaneshiro/non_interactive_test

Change action test script to default to non-interactive behavior

ikaneshiro committed:
Use Image package to extract Squashfs from SIF in image build system

Commit:
Update Go to 1.11.6

- Change Circle CI configuration to require Go 1.11.6
- Set required Go version to 1.11
- Remove checks that use version 1.10

Signed-off-by: Marcelo E. Magallon <marcelo@sylabs.io>

Commit:
Up build verbosity in CircleCI

Signed-off-by: Marcelo E. Magallon <marcelo@sylabs.io>

GodloveD committed:
update CHANGELOG.md to reflect an accurate release date

GodloveD committed:
documented some removed flags

GodloveD committed:
Merge pull request #3193 from GodloveD/CHANGELOG.md-date

CHANGELOG.md date and deprecated commands

cclerget committed:
Merge branch 'master' into macos_sandbox_img

cclerget committed:
Consolidate image package and make SIF image code more generic

cclerget committed:
Merge pull request #3177 from jmstover/macos_sandbox_img

Support Sandbox Image on macOS

cclerget committed:
Merge branch 'master' into macos_mount_tag

cclerget committed:
Merge pull request #3178 from jmstover/macos_mount_tag

Generate 9p Mount Tag on Bind Mount

mem committed:
Merge branch 'master' into issue_2534

cclerget committed:
Move partition allocation into format handlers

cclerget committed:
Merge remote-tracking branch 'upstream/master' into master-image

mem committed:
Merge branch 'master' into issue_2534

ikaneshiro committed:
Adds the old definition struct and parser back

	This was done in order to maintain compatability with the remote
	builder since recent changes to the definition struct has broken
	the builders ability to unmarshall the definition json.

ikaneshiro committed:
update formatting

cclerget committed:
Merge remote-tracking branch 'upstream/master' into master-image

cclerget committed:
Rename HasRootfs to HasRootFs

ikaneshiro committed:
fix copyright headers

Commit:
Update sif dependency to 1.0.3

- Update go.mod (and go.sum to go along with it)
- Update vendored modules
- Add minor fixes to code so that it builds with the new version

Signed-off-by: Marcelo E. Magallon <marcelo@sylabs.io>

Commit:
Run go mod tidy

Signed-off-by: Marcelo E. Magallon <marcelo@sylabs.io>

GodloveD committed:
Merge pull request #2817 from mem/issue_2534

Print usage information if an invalid flag is passed to singularity

cclerget committed:
Merge remote-tracking branch 'upstream/master' into master-caps

mem committed:
Merge branch 'master' into master-image

ikaneshiro committed:
Merge branch 'master' into rb_client_fix

ikaneshiro committed:
Merge pull request #3211 from ikaneshiro/rb_client_fix

Reintroduce old definition struct and parser for remote builds

Commit:
Merge tag 'v3.1.1'

tagging the 3.1.1 point release

cclerget committed:
Merge remote-tracking branch 'upstream/master' into master-image

cclerget committed:
Merge branch 'master-image' of github.com:cclerget/singularity into master-image

cclerget committed:
Merge remote-tracking branch 'upstream/master' into master-image

cclerget committed:
Add missing rootfs presence check

cclerget committed:
Merge pull request #3195 from cclerget/master-image

Consolidate image package and make SIF image code more generic

ikaneshiro committed:
Merge remote-tracking branch 'upstream/master' into build_unpack_refactor

ikaneshiro committed:
add check for rootfs

cclerget committed:
Merge pull request #3227 from ikaneshiro/build_unpack_refactor

Building from SIF images without priv

cclerget committed:
Merge remote-tracking branch 'upstream/master' into master-caps

cclerget committed:
Merge pull request #3187 from cclerget/master-caps

CLI capability fix

cclerget committed:
Merge remote-tracking branch 'upstream/master' into master-instance-fix

cclerget committed:
Merge pull request #3188 from cclerget/master-instance-fix

Check if an instance is not started from another one

gvallee committed:
Address Cedric's comments: better error management and display warnings when necessary. Also improve a few  comments about the usage of Seek() and so some special cases.

gvallee committed:
Address Cedric's comments: better error management and display warnings when necessary. Also improve a few  comments about the usage of Seek() and so some special cases.

gvallee committed:
Rebase to master and propagate a few changes made in sypgp.go

gvallee committed:
Fix problems after rebasing to master
